### DISCUSSION TITLE
Is it possible to create window with passing the pathname prop?

### DISCUSSION THREAD

**Discussion by baseplate-admin** [r0x0r/pywebview#1404] [Q&A]
Hi,

I have the following code:
```python
import webview


def evaluate_js(window):
    result = window.evaluate_js(
        r"""
        window.location
        """
    )

    print(result)

    try:
        result = window.evaluate_js("syntaxerror#$%#$")
    except webview.JavascriptException as e:
        print("Javascript exception occured: ", e)


if __name__ == "__main__":
    window = webview.create_window(
        "Seeder", "build/index.html", min_size=(600, 450), text_select=True
    )

    webview.start(
        evaluate_js,
        window,
        debug=True,
    )

```

Which outputs:
```ts
{'ancestorOrigins': [], 'href': 'http://127.0.0.1:54873/index.html', 'origin': 'http://127.0.0.1:54873', 'protocol': 'http:', 'host': '127.0.0.1:54873', 'hostname': '127.0.0.1', 'port': '54873', 'pathname': '/index.html', 'search': '', 'hash': ''}
```

The project is a SPA generated by [`@sveltejs/kit`](https://github.com/sveltejs/kit).

The site works fine in https://localhost:4173 when using `vite preview`. I was wondering if it was possible to configure pathname at creation time?

Many thanks.

<sub> 

link to github [repo ](https://github.com/moonlitgrace/coreproject-seeder)

</sub>

**Comment by r0x0r** [comment]
Sorry I do not follow. What pathname are you trying to configure?

**Reply by baseplate-admin** [reply]
Hi,

So basically when i create a website with `svelte-kit`, it is hosted at `https://localhost:4173`, it also has an output file called `index.html`.

So when i point that `index.html` file to `pywebview`, it hosts the index.html file at `https://localhost:12504/index.html`, where `12504` is a random number. But what i actually want `pywebview` to do is to host the `index.html` at `https://localhost:12504`.


Basically i want to do something like this:
```python
from Bottle import route,run,template,request,static_file


@route('/')
def home():
    return template('Templates/index.html',name=request.environ.get('REMOTE_ADDR'))

run(host='Work-PC',port=9999,debug=True)
```

**Reply by r0x0r** [reply]
You can specify a port pywebview's http server serves content on using  `webview.start(http_port=PORT)`. Alternatively you can use any http server you like and just point pywebview's url to it.

**Reply by baseplate-admin** [reply]
Hmm interesting. what i meant here is if i can replicate that bottle part  with `pywebview` without any external dependency.
